archunit/src/main/java/com/tngtech/archunit/core/domain/properties/HasThrowsTypes.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2018 TNG Technology Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.tngtech.archunit.core.domain.properties;
+
+import com.google.common.collect.ImmutableList;
+import com.tngtech.archunit.PublicAPI;
+import com.tngtech.archunit.base.DescribedPredicate;
+import com.tngtech.archunit.core.domain.JavaClassList;
+
+import java.util.List;
+
+import static com.tngtech.archunit.PublicAPI.Usage.ACCESS;
+import static com.tngtech.archunit.base.DescribedPredicate.equalTo;
+import static com.tngtech.archunit.core.domain.Formatters.formatMethodParameterAndThrowsTypeNames;
+import static com.tngtech.archunit.core.domain.JavaClass.namesOf;
+import static com.tngtech.archunit.core.domain.JavaClassList.GET_NAMES;
+
+public interface HasThrowsTypes {
```

Comment:
I would only introduce this interface if there is more than one reasonable class implementing it. At the moment I only see `JavaCodeUnit`. I think it would make sense to make `CodeUnitCallTarget` implement this interface, too?
The reason for the existence of `CodeUnitCallTarget` (as target of a method / constructor call) as opposed to a `JavaCodeUnit` is the same as above, a call target is not the same as a method / constructor. Because within the bytecode there will be declared a call to something like `SubFoo.bar()`, but maybe `SubFoo` doesn't declare method `bar()`, but instead it's super class `Foo` does. And maybe some interface `IFoo` declares a method `bar()`, too. So it's not immediately clear, which actual `JavaCodeUnit` belongs to a call target. Nevertheless we can already derive some things like what parameters the target has.
`CodeUnitCallTarget` offers a method `resolve()` though, that tries to find the respective method that matches the target. So we could implement `HasThrowsDeclarations` (:wink:) here, too, and pipe to `resolve().getThrowsDeclarations()`. If multiple methods match `resolve()` (the case of several interfaces declaring the same method and no class implementing it up to the part in the hierarchy that is referenced) it seems like throws declarations should be intersected (if you have interface `A` and `B` both declaring `foo`, but with different throws declarations, then `C extends A, B` can per se neither declare the one or the other checked exception)
What do you think?